tests/tests.py

```diff
@@ -30,7 +30,8 @@ def test_with_timeout(tmpfile):
     with pytest.raises(portalocker.AlreadyLocked):
         with portalocker.Lock(tmpfile, timeout=0.1) as fh:
             print('writing some stuff to my cache...', file=fh)
-            with portalocker.Lock(tmpfile, timeout=0.1, mode='wb'):
+            with portalocker.Lock(tmpfile, timeout=0.1, mode='wb',
+                                  fail_when_locked=True):
```

Comment:
When `fail_when_locked` is set to `True` the behaviour is actually different which is why we have several tests: https://github.com/TWAC/portalocker/blob/21491e43efd8b40f6de2f65b6bfd1a5a02b8b986/tests/tests.py#L28-L55